File: C#/179. Largest Number.cs
/*
 * Link: https://leetcode.com/problems/largest-number/
 * Author: Dung Nguyen Tien (Chris)
*/

public class Solution {
    public string LargestNumber(int[] nums) {
        if (EqualToZero(nums)) return "0";
        
        var convert = ConvertArray(nums);
        Array.Sort(convert, delegate(string a, string b) {
            return (b + a).CompareTo(a + b);
        });
            
        return string.Join("", convert);
    }
    
    public string[] ConvertArray(int [] nums) {
        var convert = new string[nums.Length];
        
        for(var i = 0; i < nums.Length; i++) {
            convert[i] = nums[i].ToString();
        }
        
        return convert;
    }
    
    public bool EqualToZero(int[] nums) {
        foreach(var n in nums) {
            if (n != 0) return false;
        }
        
        return true;
    }
}

File: Javascript/179. Largest Number.js
/**
 * Link: https://leetcode.com/problems/largest-number/
 * Author: Dung Nguyen Tien (Chris)
 * @param {number[]} nums
 * @return {string}
 */
var largestNumber = function(nums) {
    if (nums.join('') == 0) return "0";
    
    //convert to string array
    var convert = nums.map(val => val + "");
    convert.sort(function(a, b) {
        return (b + a) - (a + b);
    })
    
    return convert.join('');
};
